### PR TITLE
chore(deps): move prop-types to dependencies

### LINKED ISSUE #843: prop-types should be a direct dependency
First of all, I love your service!

I believe `prop-types` should be a direct dependency instead of a peer dependency. 

We use TypeScript and in TS you don't really use `prop-types` so it's a bit awkward to explicitly install `prop-types` in our project just to satisfy `react-imgix`. It would make more sense if `react-imgix` had a dependency to what ever version of `prop-types` is needed to run `react-imgix`.

What are your thoughts on this?

### PR DESCRIPTION
**Pull request by sherwinski**
TSIA

Resolves #843

### CODE REVIEW

**Comment by sherwinski** [comment]
Current dependencies on/for this PR:
* main
  * **PR #857** <a href="https://app.graphite.dev/github/pr/imgix/react-imgix/857" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: package-lock.json [hunk 1]
```diff
@@ -1,15 +1,16 @@
 {
   "name": "react-imgix",
-  "version": "9.4.0",
+  "version": "9.5.0",
```
Comment (sherwinski) [inline package-lock.json:3]: Thought it was worth calling out that the lock file was out of sync before this.

**Review by frederickfogerty — APPROVED** [review]
LGTM

**Comment by frederickfogerty** [comment]
By the way, this should probably have been a `fix` commit so that a release would have been cut

**Comment by sherwinski** [comment]
will I ever learn

**Pull request merged**